kerl

```diff
@@ -606,62 +606,138 @@ assert_build_name_unused() {
             name=$(echo "$l" | cut -d, -f2)
             if [ "$name" = "$1" ]; then
                 error "there's already a build named '$1'."
-                exit 1
+                return 1
             fi
         done <"$KERL_BASE_DIR"/otp_builds
     fi
 }
 
+lock() {
+    # $1: action - open | close
+    # $2: lock type - build | install
+    # $3: folder to act on
+
+    if [ "$1" = "close" ]; then
+        if [ -f "$3/$2.lock" ]; then
+            error "trying to $2 in $3, but lock file ($3/$2.lock) exists!"
+            exit 1
+        else
+            mkdir -p "$3"
+            touch "$3/$2.lock"
+        fi
+    elif [ "$1" = "open" ]; then
+        rm -f "$3/$2.lock"
+    fi
+}
+
+lock_close_build() {
+    # $_KERL_BUILD_DIR is global
+    lock "close" "build" "$_KERL_BUILD_DIR"
```

Comment:
i dont understand open and close - i would prefer "lock" "unlock" `lock_build` `unlock_build` etc

Reply:
No prob. I thought of lock as a noun, not a verb, but a verb works too 👍 (so, in my mind, `unlock` would be `open lock` 😄). I'll change it.

Reply:
91b3395